iroha/src/block.rs

```diff
@@ -483,6 +483,63 @@ impl VersionedCommittedBlock {
     pub fn hash(&self) -> Hash {
         self.as_inner_v1().hash()
     }
+
+    /// Returns header of valid block
+    pub const fn header(&self) -> &BlockHeader {
+        &self.as_inner_v1().header
+    }
+
+    /// Signatures that are verified with the `hash` of this block as `payload`.
+    pub fn verified_signatures(&self) -> Vec<Signature> {
+        self.as_inner_v1().verified_signatures()
+    }
+
+    /// Filters transactions by predicate
+    pub fn filter_tx_values_by_payload<'a>(
+        &'a self,
+        predicate: impl FnOnce(&'a Payload) -> bool + Copy + 'a,
+    ) -> impl Iterator<Item = TransactionValue> + 'a {
+        let block = self.as_inner_v1();
+        block
+            .rejected_transactions
+            .iter()
+            .map(|tx| (tx, tx.payload()))
+            .filter(move |(_, payload)| predicate(payload))
+            .map(|(tx, _)| tx)
+            .cloned()
+            .map(TransactionValue::RejectedTransaction)
+            .chain(
+                block
+                    .transactions
+                    .iter()
+                    .map(|tx| (tx, tx.payload()))
+                    .filter(move |(_, payload)| predicate(payload))
+                    .map(|(tx, _)| tx)
+                    .cloned()
+                    .map(VersionedAcceptedTransaction::from)
+                    .map(Into::into)
+                    .map(TransactionValue::Transaction),
+            )
+    }
+
+    /// Consumes block and returns iterator over transaction values
+    pub fn iter_tx_values(&'_ self) -> impl Iterator<Item = TransactionValue> + '_ {
```

Comment:
I think this method do the same as 'filter_tx_values_by_payload' with predicate always `true`. Do we really need it separately? Or we can call from inside `filter_tx_values_by_payload` method

Reply:
Yes, i would have 2 methods for that, as closure may not inline